商品情報編集機能

sorairo580 opened the pull request:
# WHAT
商品情報編集機能の実装
# WHY
商品情報編集機能の実装のため
・ログイン状態の出品者は、商品情報編集ページに遷移できる動画↓↓
https://gyazo.com/817f4e99d6bbad975449714208b0c552
・必要な情報を適切に入力して「更新する」ボタンを押すと、商品の情報を編集できる動画↓↓
https://gyazo.com/f5a9e9223a2c1234469cebd482d921de
・入力に問題がある状態で「変更する」ボタンが押された場合、情報は保存されず、編集ページに戻りエラーメッセージが表示される動画↓↓
https://gyazo.com/8ac893e2b2bfa85d3f218cf18bb07ce6
・何も編集せずに「更新する」ボタンを押しても、画像無しの商品にならない動画↓↓
https://gyazo.com/1514d5d5409b847475151a9ba0876b35
・ログイン状態の場合でも、URLを直接入力して自身が出品していない商品の商品情報編集ページへ遷移しようとすると、商品の販売状況に関わらずトップページに遷移する動画↓↓
https://gyazo.com/c593e59c78532b0e597195547b021b21
・ログアウト状態の場合は、URLを直接入力して商品情報編集ページへ遷移しようとすると、商品の販売状況に関わらずログインページに遷移する動画↓↓
https://gyazo.com/71bb4e0545d441d627b21f9cb350e3c1
・商品名やカテゴリーの情報など、すでに登録されている商品情報は商品情報編集画面を開いた時点で表示される動画↓↓
https://gyazo.com/6e3c730d4941ea6575135f880a0f0169

購入機能はまだ実装していません。
よろしくお願いします。

sorairo580 committed:
商品情報編集機能

コントローラーへのedit,updateアクションの定義
edit.html.erbの編集
show.html.erbの編集画面への遷移
routes.rbの編集

sorairo580 committed:
商品情報編集機能２

before_actionで自分の出品物でない物の編集はできないようにする
「もどる」ボタンは商品詳細画面に遷移するよう修正

TAKADA-yuma commented on app/controllers/items_controller.rb:
```diff
@@ -22,6 +23,19 @@ def show
   end
 
+  def edit
+    @item = Item.find(params[:id])
```
@item = Item.find(params[:id])
現状、こちらの記述が複数見られます。
これらはprivate以下にメソッド化して、beforeアクションで呼び出す形に修正しておきましょう！

複数回使用する変数をメソッド化することで可読性が上がり、改修作業が発生した際も手を加えやすくなります。

TAKADA-yuma reviewed (commented):
レビュー依頼ありがとうございます。プルリクエスト確認させていただきました。

修正点を追加でコメントさせて頂きましたので、ご確認をお願い致します。

```ruby
【次回プルリクエスト提出する際の注意点】
①修正が終わりましたら、`コミットとプッシュ`をして、`プルリクエストに修正内容が反映されているか確認`しましょう。
②既に頂いている挙動確認の画像や動画が修正範囲だった場合、最新の画像や動画の提出をお願い致します。
③googleフォームの送信をお願いいたします。
```


ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！
どうぞよろしくおねがいします。

sorairo580 committed:
商品情報編集機能３

before_action の追加

sorairo580 commented:
before_action を追加しました。
ご確認をお願いします。
画像↓↓
https://gyazo.com/a4dce63287025559b18293039abb515e

div-reviewer reviewed (commented):
sorairo580様、学習お疲れさまです。
商品編集の再レビュー依頼ありがとうございました。

`@item`を作成する動きをbeforeでまとめられましたね。素晴らしいです。
こちら`LGTM`とさせて頂きます。実装お疲れさまでした。

**●修正について●**
修正後は`再度レビュー依頼は出さなくて大丈夫です`ので、修正後、コミットとプッシュして頂けましたら、
マージ後、ローカルのmainブランチに忘れずにプルし、次の実装に取り組んで頂ければと思います。

**ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！**
今後の実装も応援しております！どうぞよろしくおねがいします。

sorairo580 committed:
商品情報編集機能3-1

・before_actionのmove_to_editとset_itemを入れ替え（先にset_itemを動かすため）
・ move_to_sessionsにより"user_signed_in? &&"は省略可

sorairo580 commented:
ありがとうございました！
